docs/MPURegionsUsageGuide.md

```diff
@@ -0,0 +1,222 @@
+
+## CONTENTS
+- [How to enable MPU](#how-to-enable-mpu)
+- [How to control MPU](#how-to-control-mpu)
+- [MPU Usages of platform](#mpu-usages-of-platform)
+- [Sample memory layout of MPU regions](#sample-memory-layout-of-mpu-regions)
+- [Appendix](#appendix)
+
+## How to enable MPU
+
+- Provided MPU is supported in a chip, the following configs need to be enabled for configuring MPU regions(for ARMv7M architecture):
+```
+	------------------------------------------------------------------------------------------------------------------------------
+	CONFIG FLAG			VALUE		DESCRIPTION		MENU PATH
+	------------------------------------------------------------------------------------------------------------------------------
+	CONFIG_ARMV7M_MPU		y		MPU support		Chip Selection -> MPU Support
+	CONFIG_ARMV7M_MPU_NREGIONS	16		Number of MPU regions	Chip Selection -> MPU Support -> Number of MPU Regions
+	------------------------------------------------------------------------------------------------------------------------------
+```
+
+- Similarly, for ARMv8M architecture, below MPU configs should be enabled.
+```
+	------------------------------------------------------------------------------------------------------------------------------
+	CONFIG FLAG			VALUE		DESCRIPTION		MENU PATH
+	------------------------------------------------------------------------------------------------------------------------------
+	CONFIG_ARMV8M_MPU		y		MPU support		Chip Selection -> MPU Support
+	CONFIG_ARMV8M_MPU_NREGIONS	8		Number of MPU regions	Chip Selection -> MPU Support -> Number of MPU Regions
+	------------------------------------------------------------------------------------------------------------------------------
+```
+
+## How to control MPU
+1. **How to add new MPU region**
+	- Logical mapping of MPU regions to their respective memory segments is done statically using below enumeration in common mpu.h header file (os/include/tinyara/mpu.h).
+	```c
+	enum MPU_REG_NUM {
+	#ifdef CONFIG_APP_BINARY_SEPARATION
+	#ifdef CONFIG_SUPPORT_COMMON_BINARY
+	#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+		MPU_REG_NUM_COM_LIB_TXT,	/* Common Binary */
+		MPU_REG_NUM_COM_LIB_RO,
+		MPU_REG_NUM_COM_LIB_DATA,
+	#else
+		MPU_REG_NUM_COM_LIB,
+	#endif
+	#endif
+	#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+		MPU_REG_NUM_APP_TXT,		/* Apps */
+		MPU_REG_NUM_APP_RO,
+		MPU_REG_NUM_APP_DATA,
+	#else
+		MPU_REG_NUM_APP,
+	#endif
+	#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
+		MPU_REG_NUM_STK,		/* Stack */
+	#endif
+	#else
+		MPU_REG_NUM_UFLASH,
+		MPU_REG_NUM_URAM,
+	#endif
+		MPU_REG_NUM_MAX
+	};
+	```
+	- The ordering of MPU regions in this enum is kept as above taking into consideration below rule:
+		- For overlapping address range between MPU regions, attributes of higher MPU region number will override those set by lower numbered MPU region.
+	- Your new MPU region should be added the enum above.
+2. **How to register MPU region to real MPU hardware**
```

Comment:
In markdown, **abc** is used for bold format, which is what I am using.
Since I am using a numbered list here, so I am not using ### here.
I think it is ok this way.